### ISSUE TITLE
bad result for he-aac v2

### ISSUE THREAD

**Issue by myfreeer** [mstorsjo/fdk-aac#69]
Version: https://github.com/mstorsjo/fdk-aac/commit/e7f36eed224b9530cf0eb7e56d6c43d3de14429e\
Platform: win64
Compiler: msvc
Frontend: https://github.com/nu774/fdkaac/commit/4571d9d76a1dd833ae4a063f52311a7a5553fe59
Audio Src: https://archive.org/download/Pnuma2005-09-01/pnuma2005-09-01d1t07.flac
Cmd Line: `flac -d pnuma2005-09-01d1t07.flac -o - | fdkaac - -m3 -p29 -o m3.aac`
Result: [m3.zip](https://github.com/mstorsjo/fdk-aac/files/1093757/m3.zip)

Edit: Probably caused by commit https://github.com/mstorsjo/fdk-aac/commit/39e13c1acbca94f562f9776e1555ced50dd0dfcd , I reverted this commit manually and  he-aac v2 goes fine.

**Comment by mstorsjo** [comment]
I tested the same setup more or less, on linux, and it seems to work fine.

A suspicion is that the binary data might be corrupted when piping, when on windows. Can you try by decoding into a wav file with `flac` separately (and please verify that it is ok), and then naming that input wav file to `fdkaac`, instead of using a pipe?

**Comment by myfreeer** [comment]
Cmd Line:
```
flac -d pnuma2005-09-01d1t07.flac
./fdkaac pnuma2005-09-01d1t07.wav -m3 -p29 -o m3.aac
```
Result: [m3.zip](https://github.com/mstorsjo/fdk-aac/files/1094091/m3.zip)

**Comment by mstorsjo** [comment]
Oh, indeed - I didn't test using the latest master version when I built the third party `fdkaac` frontend, but now I can reproduce the issue, and as you say, this commit seems to be the culprit. I'll revert that and look for a better fix for the fuzz crash.

**Comment by mstorsjo** [comment]
Fixed in a9c8cb2.

**Comment by mstorsjo** [comment]
Thanks for reporting and thanks for identifying the offending commit!
